### ISSUE TITLE
Upgrade github.com/go-errors/errors to v1.4.0

### ISSUE THREAD

**Issue by fschuetz04** [dsoprea/go-logging#3]
To enable use of the standard `errors.Is` with the errors returned by `log.Wrap` etc, `github.com/go-errors/errors` should be upgraded to v1.4.0 (and, if possible also your packages depending on this one). Do you want this? Should I submit a pull request?

**Comment by dsoprea** [comment]
Are you saying that it doesn't work now?

Please provide more of a rationale and any related commits to support it.

**Comment by fschuetz04** [comment]
Yes, it does not work with the currently used version of `github.com/go-errors/errors`. See the following example:

```go
import (
	"errors"
	"fmt"

	"github.com/dsoprea/go-exif/v3"
	jpegstructure "github.com/dsoprea/go-jpeg-image-structure/v2"
)

func main() {
	parser := jpegstructure.NewJpegMediaParser()
	ec, _ := parser.ParseFile("test.jpg")
	_, _, err := ec.Exif()
	fmt.Println("error:", err)
	fmt.Println("errors.Is:", errors.Is(err, exif.ErrNoExif))
}
```

Given a valid `test.jpg` file without EXIF data, this prints the following:

```
error: no exif data
errors.Is: false
```

However, I would expect `errors.Is` to return `true` in this case. Version `v1.4.0` of `github.com/go-errors/errors` implements `Unwrap()`  for its errors type, enabling `errors.Is` to correctly return true.

It also works by manually pinning `github.com/go-errors/errors` to `v1.4.0` or by using the `errors.Is` method of `github.com/go-errors/errors`, but those are suboptimal solutions.

**Comment by dsoprea** [comment]
It has worked in the past. Notbsure when/how this was broken in upstream.
Have you investigated why Is() doesn't work at the currently consumed
revision?



Dustin

**Comment by fschuetz04** [comment]
In the used version, the `Error` type of `github.com/go-errors/errors` doesn't define `Is` or `Unwrap`, so `errors.Is` can just check the top error. Since that is not equal to the given error, `errors.Is` returns false. In the latest version, `Unwrap` is defined, so `errors.Is` can also check the wrapped error. See https://github.com/go-errors/errors/blob/master/error.go#L206 and <https://blog.golang.org/go1.13-errors#TOC_3.2.>.

**Comment by fschuetz04** [comment]
Any updates on this? Should I investigate further?
